packages/babel-parser/src/plugins/typescript.js

```diff
@@ -1283,7 +1283,8 @@ export default (superClass: Class<Parser>): Class<Parser> =>
     ): N.Pattern | N.TSParameterProperty {
       let accessibility: ?N.Accessibility;
       let readonly = false;
-      if (allowModifiers) {
+
+      if (allowModifiers && this.lookahead().type !== tt.parenR) {
```

Comment:
Why isn't it caught by `tsNextTokenCanFollowModifier` called by `tsParseModifier?`

Reply:
good catch!